cmd/register.go

```diff
@@ -52,6 +67,31 @@ user: netclient register -s <server> -u <user_name> // attempt to join/register
 	},
 }
 
+func setHostFields(cmd *cobra.Command) {
+	fmt.Println("setting host fields")
+	if port, err := cmd.Flags().GetInt(registerFlags.Port); err == nil && port != 0 {
+		// check if port is available
+		if !ncutils.IsPortFree(port) {
+			fmt.Printf("port %d is not free\n", port)
+			os.Exit(1)
+		}
+		config.Netclient().ListenPort = port
+	}
+	if endpointIP, err := cmd.Flags().GetString(registerFlags.EndpointIP); err == nil && endpointIP != "" {
+		config.Netclient().EndpointIP = net.ParseIP(endpointIP)
+	}
+	if mtu, err := cmd.Flags().GetInt(registerFlags.MTU); err == nil && mtu != 0 {
+		config.Netclient().MTU = mtu
+	}
+	if hostName, err := cmd.Flags().GetString(registerFlags.Name); err == nil && hostName != "" {
+		config.Netclient().Name = hostName
+	}
+
+	if isStatic, err := cmd.Flags().GetBool(registerFlags.Static); err == nil {
```

Comment:
If isStatic and endpoint need to be tied together.  An endpoint should not be able to be specified is isStatic is false.  Similarily if is Static is true, endpoint should be specified.

Reply:
I don't we should make it complicated, user is well aware that endpoint might change if host is not set as static host